src/app/icd/ICDManager.cpp

```diff
@@ -167,6 +170,8 @@ void ICDManager::UpdateOperationState(OperationalState state)
         {
             uint16_t activeModeThreshold = IcdManagementServer::GetInstance().GetActiveModeThreshold();
             DeviceLayer::SystemLayer().ExtendTimerTo(System::Clock::Timeout(activeModeThreshold), OnActiveModeDone, this);
+            DeviceLayer::SystemLayer().ExtendTimerTo(System::Clock::Timeout(activeModeThreshold - ICD_ACTIVE_TIME_JITTER_MS),
```

Comment:
What Prevents our `OnActiveModeAlmostDone` to be extended/ created again when the OnActiveModeAlmostDone(); callback send messages for the sync

Reply:
We now have a boolean to prevent checking multiple time, this needs to be updated once we have unit tests to implement a way to do multiple extensions of the timer check without looping in the event of a report emission.